client/ayon_deadline/plugins/publish/global/collect_deadline_job_env_vars.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+"""Collect Deadline servers from instance.
+
+This is resolving index of server lists stored in `deadlineServers` instance
+attribute or using default server if that attribute doesn't exists.
+
+"""
+import os
+
+import pyblish.api
+
+from ayon_deadline.lib import FARM_FAMILIES
+
+
+class CollectDeadlineJobEnvVars(pyblish.api.ContextPlugin):
+    """Collect set of environment variables to submit with deadline jobs"""
+    order = pyblish.api.CollectorOrder
+    label = "Deadline Farm Environment Variables"
+    families = FARM_FAMILIES
+    targets = ["local"]
+
+    ENV_KEYS = [
+        # AYON
+        "AYON_BUNDLE_NAME",
+        "AYON_DEFAULT_SETTINGS_VARIANT",
+        "AYON_PROJECT_NAME",
+        "AYON_FOLDER_PATH",
+        "AYON_TASK_NAME",
+        "AYON_APP_NAME",
+        "AYON_WORKDIR",
+        "AYON_APP_NAME",
+        "AYON_LOG_NO_COLORS",
+        "AYON_IN_TESTS",
+        "IS_TEST",  # backwards compatibility
+
+        # Ftrack
+        "FTRACK_API_KEY",
+        "FTRACK_API_USER",
+        "FTRACK_SERVER",
+        "PYBLISHPLUGINPATH",
+
+        # Shotgrid
+        "OPENPYPE_SG_USER",
+    ]
+
+    def process(self, context):
+        env = {}
+        for key in self.ENV_KEYS:
+            value = os.getenv(key)
+            if value:
+                self.log.debug(f"Setting job env: {key}: {value}")
+                env[key] = value
+
+        # Transfer some environment variables from current context
+        context.data.setdefault("job_env", {}).update(env)
+
+
+class CollectDeadlineNukeJobEnvVars(CollectDeadlineJobEnvVars):
+    """Collect set of environment variables to submit with deadline jobs"""
+    label = "Deadline Farm Environment Variables (Nuke)"
+    hosts = ["nuke"]
+    families = ["render", "prerender"]
+
+    ENV_KEYS = [
+        # Backwards compatibility
+        "PYTHONPATH",
+        "PATH",
+    ]
+
+
+class CollectAYONServerUrlToFarmJob(CollectDeadlineJobEnvVars):
+    label = "Submit AYON server URL to farm job"
+
+    # TODO: Expose the enabled/optional state to settings
+    ENV_KEYS = [
+        "AYON_SERVER_URL",
+        "AYON_USE_DEV",
+    ]
```

Comment:
### Review only this code if doing code review

**NOTE** 

This is the ONLY related code changes for this PR - everything else is also in #32

Reply:
Do we want send AYON_SERVER_URL always? If thats the case, server url in AYON Deadline plugin doesn't make sense, neither code for it in `GlobalJobPreload`.

(Just note, API keys for all DL servers would need to be same as they would be still set via AYON DL plugin.)

Reply:
> Do we want send AYON_SERVER_URL always? If thats the case, server url in AYON Deadline plugin doesn't make sense, neither code for it in GlobalJobPreload.

No - I think we'd want to make this plug-in disabled by default, and configurable via settings to enable it.

> (Just note, API keys for all DL servers would need to be same as they would be still set via AYON DL plugin.)

Sir - that's a **very valid** point. We may need to submit that along too. @tweak-wtf, thoughts?

Reply:
oh sheet, true! thanks for catching!
i really like what's described in https://github.com/ynput/ayon-documentation/pull/285
seems like minimal impact on the existing addon codebase and on the additional AYON instances settings.

if we want to expand on that and the n:n idea between AYON and DL, i could imagine the AYON Deadline Plugin Settings getting either a new section `Additional AYON Servers` or a general list of dicts kinda like the current AYON server settings for the DL servers.
then submitting the current AYON URL could become enabled by default. whether that's desirable, i can't really say.

for me using `AYON_API_KEY` in `ayon+settings://core/environments` as an override in that instance's settings really does the trick.

Reply:
How does that work, with the `AYON_API_KEY` in settings? When `GlobalJobPreLoad` triggers the connection with the AYON server - it already needs a valid API key to make the connection? Seems like for you it may just happen to work because it maybe found some matching 'login' using your local cached login?